examples/where+regression/where+regression.ipynb

```diff
@@ -115,167 +116,88 @@
-      "  _C._check_onnx_proto(proto)\n"
+      "/Users/mhchia/projects/work/pse/zk-stats-lib/zkstats/computation.py:166: TracerWarning: torch.tensor results are registered as constants in the trace. You can safely ignore this warning if you use this function to create tensors out of constant variables that would be the same every time you call this function. In any other case, this might cause the trace to be incorrect.\n",
+      "  is_precise_aggregated = torch.tensor(1.0)\n",
+      "/Users/mhchia/Library/Caches/pypoetry/virtualenvs/zkstats-brXmXluj-py3.12/lib/python3.12/site-packages/torch/onnx/symbolic_opset9.py:2174: FutureWarning: 'torch.onnx.symbolic_opset9._cast_Bool' is deprecated in version 2.0 and will be removed in the future. Please Avoid using this function and create a Cast node instead.\n",
+      "  return fn(g, to_cast_func(g, input, False), to_cast_func(g, other, False))\n"
      ]
     }
    ],
    "source": [
-    "# Verifier/ data consumer side:\n",
-    "# Want to calculate regression of y over\n",
-    "class verifier_model(nn.Module):\n",
-    "    def __init__(self):\n",
-    "        super(verifier_model, self).__init__()\n",
-    "        self.w = nn.Parameter(data = lt20_dummy_w_tensor, requires_grad = False)\n",
+    "# Verifier/ data consumer side: send desired calculation\n",
+    "from zkstats.computation import computation_to_model, State\n",
     "\n",
-    "    def forward(self, *args):\n",
-    "        # infer Y from the last parameter\n",
-    "        Y = args[-1]\n",
-    "        filter = (Y<20)\n",
-    "        X_one = torch.cat((*args[:-1], torch.ones_like(args[0])), dim=2)\n",
-    "        fil_X_one = torch.where(filter, X_one, torch.tensor([0.0]*X_one.size()[2]))\n",
-    "        fil_X_T = torch.transpose(fil_X_one, 1, 2)\n",
-    "        fil_Y = torch.where(filter, Y, 0.0)\n",
     "\n",
-    "        return (\n",
-    "            torch.sum(torch.abs(fil_X_T @ fil_X_one @ self.w - fil_X_T @ fil_Y)) <= 0.01 * torch.sum(torch.abs(fil_X_T @ fil_Y)),\n",
-    "            self.w\n",
-    "        )\n",
+    "def computation(s: State, data: list[torch.Tensor]) -> torch.Tensor:\n",
+    "    x = data[0]\n",
+    "    y = data[1]\n",
+    "    # FIXME: should be replaced by `s.where` when it's available. Now the result may be incorrect\n",
+    "    filter = (y < 20)\n",
+    "    # FIXME: not sure how to do filtering correctly here\n",
+    "    filtered_x = torch.where(filter, x, 0.0)\n",
```

Comment:
This should be wrong for calculating `filtered_x` but I'm not sure how to do it since in the example `torch.where` is done after `torch.cat`. Do you have any thoughts on this, or I'm thinking we can just leave it for now and change it after `State.where` is supported?

Reply:
Thx Kevin! Nw, will open a new PR that specifically address where op!